model/buildlogger.go

```diff
@@ -464,87 +451,75 @@ func (l *Logs) Find(ctx context.Context, opts LogFindOptions) error {
 	} else {
 		return mongo.ErrNoDocuments
 	}
+	if opts.LatestExecution {
+		l.filterLatestExecution()
+	}
 
 	return nil
 }
 
+func (l *Logs) filterLatestExecution() {
+	execution := 0
+	for i, log := range l.Logs {
+		if log.Info.Execution < execution {
+			l.Logs = l.Logs[:i]
+			break
+		} else {
+			execution = log.Info.Execution
+		}
+	}
+}
+
 func createFindQuery(opts LogFindOptions) map[string]interface{} {
 	search := bson.M{
 		logCreatedAtKey:   bson.M{"$lte": opts.TimeRange.EndAt},
 		logCompletedAtKey: bson.M{"$gte": opts.TimeRange.StartAt},
 	}
 	if opts.Info.Project != "" {
 		search[bsonutil.GetDottedKeyName(logInfoKey, logInfoProjectKey)] = opts.Info.Project
-	} else if opts.Empty.Project {
```

Comment:
Are all of these empty field options no longer necessary (aside from test name)?

Reply:
No that's why I removed them